### ISSUE TITLE
Filter with the '==' operator

### ISSUE THREAD

**Issue by luissalgadofreire** [caesar0301/treelib#62]
Great library, to start with. I really find it useful.

One quick doubt. Is there a straightforward way to search for nodes with a specific tag instead of by nid?

Either using `tree.show()` or `tree.expand_tree()` with a filter, everything works fine if one uses something like:

```
nodes = tree.expand_tree('root', filter=lambda x: x.tag != 'ART')
for node in nodes: print(node)
```

But if we use the equality operator, nothing is returned.

```
nodes = ctree.expand_tree('root', filter=lambda x: x.tag == 'ART')
for node in nodes: print(node)
```

Looking at the source code, I can kind of understand why. But is there a canonical way of searching nodes by tag instead of by nid? The idea is to easily get all nodes with a given tag.

Thanks for the great library.

**Comment by caesar0301** [comment]
Hmm, interesting. U let me thinks of XPath in xml/html parsers, although these are high-level libraries on tree structure. Maybe we can add this basic routine like XPath. e.g.
```
ctree.find("/rootTag/*/ART")  # Find all nodes with ART tag of root
ctree.find("nodeTag/ART") # Find all children nodes with ART tag of specific node
```
This seems great!

**Comment by luissalgadofreire** [comment]
Sounds great and straight to the point. It would even serve more use cases than I was thinking.

I'm a noob in programming and python but ended up simply doing:

def find(self, tag):
        nodes = []
        all_nodes = self.all_nodes()
        for node in all_nodes:
            if node.tag == tag:
                nodes.append(node)
        return nodes

I suspect, though, there might be more performant ways of achieving this.

**Comment by caesar0301** [comment]
Yes, I am afraid u need a node traverse by hand for now to filter your interested nodes. You can get a more efficient method to use python [] operator.
```
nodes = [n for n in tree.all_nodes() if n.tag == YOUR_TAG]
```
PS. thanks for you advise of tag and data. I expect to unify their grammar into the same style.
👍

**Comment by luissalgadofreire** [comment]
Cool. I'll do as you suggest.

Thanks for the heads up.

**Comment by BebeSparkelSparkel** [comment]
Hey everyone. In pull request #63 I added the filter_nodes methods. It's not a perfect solution but you can use `tree.filter_nodes(lambda n: n.tag=='tag string')`.

You can also use this for data `tree.filter_nodes(lambda n: n.data==data_object)`.

I was having the same problem and this seems to work. Also there is a working example of this in issue #65.

**Comment by liamlundy** [comment]
Closed by #63
